spec/rspec/mocks/any_instance_spec.rb

```diff
@@ -1225,6 +1242,14 @@ def dup(_)
           expect(instance.existing_method).to eq :existing_method_return_value
         end
 
+       it "does not restore a stubbed method not originally implemented in the class" do
+          allow_any_instance_of(::AnyInstanceSpec::ChildClass).to receive(:foo).and_return :result
+          expect(::AnyInstanceSpec::ChildClass.new.foo).to eq :result
+
+          reset_all
```

Comment:
Thanks @myronmarston, `reset_all` is what I needed... I was able to drop the extra example and guarantee the test failure on ruby 2.3 without the `recorder.rb` changes.  :+1: 

I moved the example to an existing context `...after the test has finished` that seems to be describe what we're doing here.  The test output looks much better now, thanks!

```
  when used after the test has finished
    restores the original behaviour, even if the expectation fails
    does not restore a stubbed method not originally implemented in the class
    restores the original behavior of a stubbed method
```